moto/dynamodb2/models.py

```diff
@@ -1123,6 +1126,14 @@ def _trim_results(self, results, limit, exclusive_start_key, scanned_index=None)
                     break
 
         last_evaluated_key = None
+        size_limit = 1000000  # DynamoDB has a 1MB size limit
```

Comment:
How do we know it is 1000*1000 and not 1024*1024? Was there official documentation or a test done against real DDB?

Reply:
I've just done some testing against a real DDB. Querying the same data gives me 16913 items back in one go, with 60 bytes per item.
16913 * 60 = 1014780 - i.e., some weird number in between 1000000 and 10241024.

I've noticed a similar discrepancy when implementing the 400KB item limit. DDB will happily accept an item with size 402KB, but it will only start complaining around 404/405KB?

Not sure why, but it looks like they have some soft limits where you can go slightly over, but not by much.

Reply:
Hi guys ! :). I'm impressed with this behavior 😮 . @bblommers , I would like to test this behavior by myself due some personal/job related projects. Did you try this with your AWS account or with the dockerized version of DDB ?